.github/workflows/ci.yaml

```diff
@@ -0,0 +1,35 @@
+on:
+  push:
+    branches: [ staging, trying, master ]
+  pull_request:
+
+name: Continuous integration
+
+jobs:
+  ci-linux:
+    runs-on: ubuntu-20.04
+    continue-on-error: ${{ matrix.experimental || false }}
+    strategy:
+      matrix:
+        # All generated code should be running on stable now, MRSV is 1.59.0
+        rust-version: [nightly, stable, 1.59.0]
+        toolchain: [riscv32i-unknown-none-elf, riscv32imc-unknown-none-elf, riscv32imac-unknown-none-elf, riscv64imac-unknown-none-elf, riscv64gc-unknown-none-elf]
```

Comment:
Could you also add `x86_64-unknown-linux-gnu` to the list?

Reply:
Wouldn't this always fail? I think the inline asm would prevent it from compiling for x86_64. Do you mean `riscv64gc-unknown-linux-gnu`?

Reply:
Hmm, indeed, it fails currently. The idea behind this was to make sure documentation builds successfully and iirc `x86_64-unknown-linux-gnu` is the default target used on docs.rs. `cortex-m-semihosting` in contrast builds successfully for the host target.

Reply:
I figured it out; it's because those crates only compile the inline asm if the target architecture is correct, otherwise it uses `unimplemented!()`, meaning that the crate can be built for any architecture. I could add this, but I fail to see the benefit. It would be better to assert the correct architecture at compile time; users working on multiple architectures would have to conditionally compile this library anyways. 

For testing the documentation, I can add a run of `cargo doc`.